app/models/item.rb

```diff
@@ -1,3 +1,9 @@
 class Item < ApplicationRecord
   belongs_to :supermarket
+  has_many :customer_items
+  has_many :customers, through: :customer_items
```

Comment:
Many to many setup looks excellent! Also, well done encapsulating this logic for counting in the model! Even though it's a simple calculation, it's most MVC-adherent to put it here :)

Reply:
Make sure to add model tests! For all your relationships and for this one instance method